discovery-provider/src/model_validator.py

```diff
@@ -0,0 +1,136 @@
+    def get_schema(cls, field, model):
+        # If model is not present in dict, init its schema and its field subschemas
+        if model not in cls.models_to_schema_and_fields_dict:
+            cls.init_model_schemas(model)
+
+        # If field is empty, return the entire model schema
+        if field == "":
+            return cls.models_to_schema_and_fields_dict[model]['model_schema']
+
+        # Else, return the specified field schema
+        return cls.models_to_schema_and_fields_dict[model]['field_schema'][field]
+
+    @classmethod
+    def init_model_schemas(cls, model):
+        # Load in the model schema in /schemas
+        model_filename = './src/schemas/schema_' + model.lower() + '.json'
+        with open(model_filename) as f:
+            schema = json.load(f)
+
+        model_properties = {
+            "model_schema": schema, # schema for the entire model
+            "field_schema": {}, # schema for just a field in model
+            "fields": schema['definitions'][model]['required'] # list of fields in model
+        }
+
+        cls.models_to_schema_and_fields_dict[model] = model_properties
+
+        # Create all the subschemas for each individual field
+        for field in model_properties['fields']:
+            # Create a deep copy of the entire schema
+            field_schema = copy.deepcopy(schema)
+
+            # Replace the properties value with just the field to validate against
+            # This way, we are able to use one entire model schema and at runtime,
+            # generate a new schema for just a field
+            # ex. replace all properties of Track (blockhash, block, title, ...) with just 'title'
+            field_to_validate_against = {field: field_schema['definitions'][model]['properties'][field]}
+            field_schema['definitions'][model]['properties'] = field_to_validate_against
+            field_schema['definitions'][model]['required'] = [field]
+
+            # Add field schema to dict
+            cls.models_to_schema_and_fields_dict[model]['field_schema'][field] = field_schema
+
+        return schema
+
+    # TODO: add another field in the class dict called 'default' and 'type' for each field so
+    # there is no need to dig down the subschema for these properties
+    
+    @classmethod
+    def get_field_default(cls, model, field):
+        if model in cls.models_to_schema_and_fields_dict:
+            field_schema = cls.models_to_schema_and_fields_dict[model]['field_schema'][field]
+            return field_schema['definitions'][model]['properties'][field]['default']
+        return None
+
+    @classmethod
+    def get_field_type(cls, model, field):
+        if model in cls.models_to_schema_and_fields_dict:
+            field_schema = cls.models_to_schema_and_fields_dict[model]['field_schema'][field]
+            return field_schema['definitions'][model]['properties'][field]['type']
+        return None
+
+# Creating a track with improper title and download fields
+'''
```

Comment:
curious what these are for? documentation sake? maybe put at the top in the docstring? not super familiar w/ this convention, but maybe ok?

Reply:
these were for the sake of the PR in giving context of what the logs would look like with example inserts. i don't plan on keeping these in the final PR